analyze_map.go

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package crossplane
+
+import "fmt"
+
+// mapParameterMasks holds bit masks that define the behavior of the body of map-like directives.
+// Some map directives have "special parameter" with different behaviors than the default.
+type mapParameterMasks struct {
+	specialParameterMasks map[string]uint
+	defaultMasks          uint
+}
+
+//nolint:gochecknoglobals
+var mapBodies = map[string]mapParameterMasks{
+	"charset_map": {
+		defaultMasks: ngxConfTake1,
+	},
+	"geo": {
+		specialParameterMasks: map[string]uint{"ranges": ngxConfNoArgs, "proxy_recursive": ngxConfNoArgs},
```

Comment:
can you add test case(s) for proxy_recursive?

Reply:
Added!